### PR TITLE
#159965308 Comment article 

### PR DESCRIPTION
**Pull request by mikenthiwa**
#### What does this PR do?
- Allows users to comment on articles.

#### Description of task to be completed?
- A user is able to add, get and delete a comment.

#### How should this be manually tested?
$ git clone https://github.com/andela/ah-magnificent6.git
$ git checkout ft-comment-article-159965308
$ cd ah-magnificent6
$ cp .env-sample .env [and set the environment variables]
$ python3 -m venv venv
$ source venv/bin/activate
(venv) $ pip install -r requirements.txt
(venv) $ python manage.py migrate
(venv) $ python manage.py runserver

Using [Postman](https://www.getpostman.com/) navigate to 
User registration: `POST {your-server-root}:8000/api/user/signup`

```
{
    "email": "user1@user.user",
    "username": "user1",
    "password": "User$#@!"
}
```

User Login: `POST {your-server-root}:8000/api/user/login`
```
{
    "email": "user1@user.user",
    "password": "User$#@!"
}

```
Use the token from the response in the login as the authentication header while creating, retrieving and deleting a comment.
**Create a comment**
`POST
http://127.0.0.1:8000/api/articles/slug/comments/
`
screen shot 2018-09-21 at 11 08 37

**Get a comment**
`GET
http://127.0.0.1:8000/api/articles/slug/comments/comment_id/
`
screen shot 2018-09-24 at 15 51 05

**Delete a comment**
`DELETE
http://127.0.0.1:8000/api/articles/slug/comments/comment_id/
`

### CODE REVIEW

File: authors/apps/articles/views.py [hunk 1]
```diff
@@ -411,8 +424,55 @@ def post(self, request, slug):
...
+        # query the database to get article with provided slug
+        article = Article.objects.get(slug=slug)
+
+        comment_data = {
+            'commented_by': request.user.username,
+            'comment_body': request.data.get('comment_body', None)
+        }
+        created_comment = Comment.objects.create(
+            commented_by=request.user, article=article, comment_body=request.data.get('comment_body', None))
+
```
Comment (Bileonaire) [inline authors/apps/articles/views.py]: Kindly fix the indentation here.

File: authors/apps/articles/tests/test_comments.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+        self.article_url = reverse('articles:create')
+        self.register_user = self.client.post(self.registration_url, self.user_data)
+        token = generate_jwt_token(self.user_data['username'])
+        self.client.get(reverse("authentication:activate_user", args=[token]))
+        login_response = self.client.post(self.login_url, self.user_data, format='json')
+        self.headers = {'HTTP_AUTHORIZATION': 'Bearer {}'.format(login_response.data.get('token'))}
+        self.slug = self.client.post(self.article_url, self.article_data, format="json", **self.headers)
+        self.comments_url = reverse('articles:comments', kwargs={"slug": self.slug.data['slug']})
+        self.detail_comment = reverse('articles:comment_detail', kwargs={"slug": self.slug.data['slug'], "pk": 2})
```
Comment (Bileonaire) [inline authors/apps/articles/tests/test_comments.py:36]: Kindlly fix the indentation here

**Review by Bileonaire — CHANGES_REQUESTED** [review]

File: authors/apps/articles/views.py (hunk 1 above)
Comment (mikenthiwa) [reply authors/apps/articles/views.py]: @Bileonaire Done

**Comment by micahoriaso** [comment]
screen shot 2018-09-24 at 10 55 15
@mikenthiwa please give the user a descriptive response when there are no comments to show.

**Comment by micahoriaso** [comment]
screen shot 2018-09-24 at 11 01 29
When the user tries to comment on a non existent article this is the error he/she gets. Please try to capture this and give the user an appropirate message.

**Comment by pngg** [comment]
Looks good to me just needs the rebasing.

File: authors/apps/articles/tests/test_comments.py (hunk 2 above)
Comment (mikenthiwa) [reply authors/apps/articles/tests/test_comments.py:36]: @Bileonaire done

**Review by Bileonaire — APPROVED** [review]

**Pull request merged**